u-boot control DTB for the rock-pi-4c: enable an SPI TPM on spi1 for measured
boot.

  - keep an OP-TEE firmware node (linaro,optee-tz): OP-TEE stays loaded and is
    used by RNG_OPTEE;
  - drive spi1 CS from GPIO4_D5 (the native spi1 CS is claimed by hardware and
    unusable here), disable the unpopulated NOR flash sharing spi1 CS0, and add
    a TPM TIS SPI device (tcg,tpm_tis-spi).

With the TPM visible, u-boot exposes EFI_TCG2_PROTOCOL so systemd-stub measures
the UKI into PCR 11.

--- a/arch/arm/dts/rk3399-u-boot.dtsi	2026-08-10 14:18:55.270633254 +0100
+++ b/arch/arm/dts/rk3399-u-boot.dtsi	2026-08-10 14:32:07.271917410 +0100
@@ -16,6 +16,13 @@
 		u-boot,spl-boot-order = "same-as-spl", &sdhci, &sdmmc;
 	};
 
+	firmware {
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+	};
+
 	pmusgrf: syscon@ff330000 {
 		compatible = "rockchip,rk3399-pmusgrf", "syscon";
 		reg = <0x0 0xff330000 0x0 0xe3d4>;
--- a/arch/arm/dts/rk3399-rock-pi-4c-u-boot.dtsi	2026-08-10 14:18:55.272633285 +0100
+++ b/arch/arm/dts/rk3399-rock-pi-4c-u-boot.dtsi	2026-08-10 14:32:07.272105914 +0100
@@ -6,8 +6,20 @@
 #include "rk3399-rock-pi-4-u-boot.dtsi"
 
 &spi1 {
+	/* The rk3399 native spi1 chip-select is claimed by hardware and
+	 * unusable on this board, so drive CS from GPIO4_D5 instead. */
+	pinctrl-0 = <&spi1_clk &spi1_tx &spi1_rx>;
+	num-cs = <1>;
+	cs-gpios = <&gpio4 RK_PD5 GPIO_ACTIVE_LOW>;
+
+	/* Unpopulated NOR flash footprint shares spi1 CS0. */
 	flash@0 {
-		bootph-pre-ram;
-		bootph-some-ram;
+		status = "disabled";
+	};
+
+	tpm@0 {
+		compatible = "tcg,tpm_tis-spi";
+		reg = <0>;
+		spi-max-frequency = <1000000>;
 	};
 };
